### ISSUE TITLE
Feature Request: Allow drag and drop of open windows.

### ISSUE THREAD

**Issue by iTerminate** [mzur/gnome-shell-wsmatrix#60]
When looking at the preview off all workspaces (super+w). Allow the user to drag and drop workspaces around as well as drag individual windows within the workspace to another workspace.

**Comment by ebeem** [comment]
I would personally love to see that. However, this feature requires a lot of time and effort.
You can take a look at different solutions/approaches we collected here https://github.com/mzur/gnome-shell-wsmatrix/issues/13#issuecomment-487524588.

I can't see this being implemented soon, but I hope we will have it in the future.

**Comment by mzur** [comment]
I agree. It would be best if this could be implemented without adding a lot of new code since this is not a very important feature. Maybe logic can be shared with the existing DnD feature of the activity overview.

**Comment by set-soft** [comment]
> I agree. It would be best if this could be implemented without adding a lot of new code since this is not a very important feature. Maybe logic can be shared with the existing DnD feature of the activity overview.

I see it really important ;-)
I used https://github.com/passingthru67/workspaces-to-dock that now isn't maintained. And was looking for a replacement. This switcher seems to be an interesting replacement, but it lacks:
1. Window DnD to move them easilly from one workspace to another.
2. Name of the workspace in addition to the thumbnail.

**Comment by mzur** [comment]
This could actually be implemented with the new GNOME 40 version of this extension, I think. If I open the workspace overview popup with <kbd>Super</kbd>+<kbd>W</kbd> I can already drag windows around. We "just" have to implement the drop action.
